app/uk/gov/hmrc/exports/models/declaration/ExportsDeclaration.scala

```diff
@@ -47,10 +50,37 @@ case class ExportsDeclaration(
 
 object ExportsDeclaration {
 
-  object REST {
+  def apply(id: String, eori: Eori, declarationRequest: ExportsDeclarationRequest): ExportsDeclaration =
```

Comment:
I like the idea of reversing the responsibility of type conversion 👍 The code is more readable on client's side as well.